velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -408,4 +408,70 @@ struct DayOfYearFunction {
     result = getDayOfYear(getDateTime(date));
   }
 };
+
+static const folly::F14FastMap<std::string, int8_t> kDayOfWeekNames{
+    {"th", 0},       {"fr", 1},     {"sa", 2},       {"su", 3},
+    {"mo", 4},       {"tu", 5},     {"we", 6},       {"thu", 0},
+    {"fri", 1},      {"sat", 2},    {"sun", 3},      {"mon", 4},
+    {"tue", 5},      {"wed", 6},    {"thursday", 0}, {"friday", 1},
+    {"saturday", 2}, {"sunday", 3}, {"monday", 4},   {"tuesday", 5},
+    {"wednesday", 6}};
+
+template <typename T>
+struct NextDayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Date>* /*startDate*/,
+      const arg_type<Varchar>* dayOfWeek) {
+    if (dayOfWeek != nullptr) {
+      weekDay_ = getDayOfWeekFromString(*dayOfWeek);
+      if (!weekDay_.has_value()) {
+        invalidFormat_ = true;
+      }
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Date>& result,
+      const arg_type<Date>& startDate,
+      const arg_type<Varchar>& dayOfWeek) {
+    if (invalidFormat_) {
+      return false;
+    }
+    auto weekDay = weekDay_.has_value() ? weekDay_.value()
+                                        : getDayOfWeekFromString(dayOfWeek);
+    if (weekDay.has_value()) {
```

Comment:
same here; return early

please, update documentation to clarify that this function returns NULL if dayOfWeek is not valid

Reply:
@mbasmanova done. Thanks for your suggestion.